rfi_file_monitor/operations/dummy_operation.py

```diff
@@ -129,9 +135,13 @@ def preflight_check(self):
     def run(self, file: File):
         if isinstance(file, Directory) and self.params.enable_directories:
             return add_directory_support(type(self)._run)(self, file)
-        return self._run(file)
+        return self._run(file, self.index)
+
+    @classmethod
+    def _attach_metadata(cls, file: File, operation_index:int):
+        file.operation_metadata[operation_index] = {'number':random.random()}
 
-    def _run(self, file: File):
+    def _run(self, file: File, operation_index):
```

Comment:
`_run` is an object method, so you can call `self.index` from within it, no need to pass it as an argument.